rest-api/bareos-restapi.py

```diff
@@ -0,0 +1,1796 @@
+):
+    """
+    Read all jobdef resources. Built on console command _show profiles_.
+
+    Needs at least Bareos Version >= 20.0.0
+    """
+    return show_configuration_items(response=response, current_user=current_user, itemType="profiles", byName=profiles_name, verbose=verbose)
+
+
+
+@app.post("/configuration/profiles", tags=["users", "configuration"])
+def post_profile(
+    *,
+    profileDef: profileResource = Body(..., title="profile resource"),
+    response: Response,
+    current_user: User = Depends(get_current_user),
+):
+    """
+    Create a new Bareos profile resource.
+    Console command used: _configure add profile_
+
+    """
+    return configure_add_standard_component(
+        response=response,
+        componentDef=profileDef,
+        componentType="profile",
+        current_user=current_user,
+    )
+
+
+@app.get("/configuration/consoles", tags=["consoles", "configuration"])
+def read_all_consoles(
+    *, response: Response, current_user: User = Depends(get_current_user), verbose: Optional[bareosBool] = Query("yes", title="Verbose output")
+):
+    """
+    Read all jobdef resources. Built on console command _show consoles_.
+
+    Needs at least Bareos Version >= 20.0.0
+    """
+    return show_configuration_items(response=response, current_user=current_user, itemType="consoles", verbose=verbose)
+
+@app.get("/configuration/consoles/{consoles_name}", tags=["consoles", "configuration"])
+def read_console_by_name(
+    *, response: Response, current_user: User = Depends(get_current_user), consoles_name: str = Path(...,title="Console name to look for"), verbose: Optional[bareosBool] = Query("yes", title="Verbose output")
+):
+    """
+    Read all jobdef resources. Built on console command _show consoles_.
+
+    Needs at least Bareos Version >= 20.0.0
+    """
+    return show_configuration_items(response=response, current_user=current_user, itemType="consoles", byName=consoles_name, verbose=verbose)
+
+
+
+@app.post("/configuration/consoles", tags=["users", "configuration"])
+def post_console(
+    *,
+    consoleDef: consoleResource = Body(..., title="console resource"),
+    response: Response,
+    current_user: User = Depends(get_current_user),
+):
+    """
+    Create a new Bareos console resource.
+    Console command used: _configure add console_
+
+    """
+    return configure_add_standard_component(
+        response=response,
+        componentDef=consoleDef,
+        componentType="console",
+        current_user=current_user,
+    )
+
+
+### Director
+
+
+@app.get("/control/directors/version", tags=["directors", "control"])
+def read_director_version(
+    *, response: Response, current_user: User = Depends(get_current_user)
+):
+    """
+    Read director version. Command used: _version_
+    """
+    result = None
+    listCommand = "version"
```

Comment:
I'm not sure if this should be named listCommand. It's a read operation but a general DIR command is used instead of the (l)list command.

Reply:
done